### PR TITLE
Block data transfer LDR

### PR DESCRIPTION
**Pull request by guerinoni**
- Memory: Add unused chunk of memory
- ARM: Add LDR in block data transfer

### CODE REVIEW

**Review by FedericoBruzzone — APPROVED** [review]
LGTM

File: emu/src/arm7tdmi.rs [hunk 1]
```diff
@@ -152,38 +152,32 @@ impl Arm7tdmi {
...
+                arm.memory.write_at(address, value.get_bits(0..=7) as u8);
+                arm.memory
+                    .write_at(address + 1, value.get_bits(8..=15) as u8);
+                arm.memory
+                    .write_at(address + 2, value.get_bits(16..=23) as u8);
+                arm.memory
+                    .write_at(address + 3, value.get_bits(24..=31) as u8);
+            };
+
+            self.exec_data_trasfer(reg_list, pre_post, &mut address, up_down, transfer);
```
Comment (AlessioC31) [inline emu/src/arm7tdmi.rs:180]: What do you think about moving this logic to `InternalMemory`? Maybe changing the trait `IoDevice`. We could introduce functions like `write_32_bits` `read_32_bits` or change the current `write_at` and `read_at` to support the amount of bytes to read/write.

Comment (guerinoni) [reply emu/src/arm7tdmi.rs:180]: I already thought on it, I think I will do in separated PR, I'm only thinking cleanest solution to do so :)

Comment (AlessioC31) [reply emu/src/arm7tdmi.rs:180]: Fair enough, can we create an issue so we don't forget? :D

Comment (guerinoni) [reply emu/src/arm7tdmi.rs:180]: I can do it :)

**Pull request merged**